src/proxy/inbound.rs

```diff
@@ -470,7 +471,10 @@ impl crate::tls::ServerCertProvider for InboundCertProvider {
             %identity,
             "fetching cert"
         );
```

Comment:
This is probably a bit easier to read if we just do `IF identity(from sock IP) == self.local_workload.get_identity() then self.local_workload.fetch_certificate() ELSE error`